Added evaluation folder and core files

BreckoD opened the pull request:
Added 'eval' folder with 'eval_runner.py' script and 'analyst_eval.ipynb' main file.

BreckoD committed:
Create eval folder and eval_runner.py

Create an evaluation folder with the evaluation launcher script eval_runner.py

BreckoD committed:
Add Analyst_eval.ipynb

Added the main file for the STTN Analyst evaluation

BreckoD committed:
Rename Analyst_eval.ipynb to analyst_eval.ipynb

BreckoD committed:
Update eval_runner.py

Rename the main file to analyst_eval.ipynb

BreckoD committed:
Add console output for results 

Add console output for results and minor output tweaks

yuribogomolov commented on src/sttn/eval/eval_runner.py:
```diff
@@ -0,0 +1,159 @@
+import nbformat
```
Does your notebook consume this script? We probably don't need to check it in now if it's not working.

BreckoD replied:
Yes, the runner script needs this library to run the notebook

yuribogomolov commented on src/sttn/eval/analyst_eval.ipynb:
```diff
@@ -0,0 +1,576 @@
+{
```
Could you please move the logic to Python modules & classes?
Notebooks are not the best interface to manage Python code. Ideally, the runner notebook should contain only a few lines of code to start the evaluation.

BreckoD replied:
Done! The new version has all evaluators in the separate file

yuribogomolov commented on src/sttn/eval/analyst_eval.ipynb:
```diff
@@ -0,0 +1,576 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "test_dataset = client.read_dataset(dataset_name=dataset_name)\n",
```
Where are the datasets stored? Can we include them in this PR?

BreckoD replied:
They are stored in LangSmith

BreckoD committed:
Merge branch 'yuribogomolov:main' into patch-1

BreckoD committed:
Merge branch 'yuribogomolov:main' into patch-1

BreckoD committed:
Merge branch 'yuribogomolov:main' into patch-1

BreckoD committed:
Adjusted to the newest eval structure

- Added LLM-based metrics (temporal and geospatial awareness);
- Fixed issue with RAM overflow, added temporary variable deletion;
- Added new metrics (categories-based) and deleted(complexities) / renamed(bad-grammar now is poorly-written) old ones;
- Improved previous evaluators and moved all of them to a separate file;
- Added GPT-4o-mini compatibility;
- Generated and added 100 examples dataset;
